sdk/assemblyscript/examples/agents/assembly/counterAgent.ts

```diff
@@ -105,10 +97,22 @@ export class CounterAgent extends Agent {
         } else {
           this.count++;
         }
+
+        // publish an event to subscribers
+        this.publishEvent(new CountUpdated(this.count));
+
         return this.count.toString();
       }
     }
 
     return null;
   }
 }
+
+
+@json
+class CountUpdated extends AgentEvent {
+  constructor(public count: i32) {
+    super("countUpdated");
+  }
+}
```

Comment:
If we wanted to hook into the event/subscription - would this interface be a good place to do it? For example: 

```suggestion
@json
class CountUpdated extends AgentEvent {
  constructor(public count: i32) {
    super("countUpdated");
  }
  
  onSubscribe(request) {
       // check request, ensure they are an admin and can read these events
  }
  
  beforeYield(data, request) {
      // format the payload depending on the requester. 
  }
}
```

Reply:
that wouldn't be possible with the current architecture.  But I'll keep the scenario in mind for the future.  Thanks.

Reply:
In terms of security, I believe this will help big time, say we have one-one agent per user in our application, what if another user obtains the other agent id?

Checking the header in something like: `onSubscribe(request) {` we can enforce things. 

```js
// hypothetical
onSubscribe(request) {
 const header = request.header.auth
 if(header.userId !=== this.agent.userId) throw new Error("brr can't do that");
````